In [12]:
%run iJO1366_V0.ipynb

Note: you may need to restart the kernel to use updated packages.
Maximize
1.0*BIOMASS_Ec_iJO1366_core_53p95M - 1.0*BIOMASS_Ec_iJO1366_core_53p95M_reverse_5c8b1


To find out the initial (baseline) Media:

In [13]:
model.medium

{'EX_co2_e': 1000.0,
 'EX_cobalt2_e': 1000.0,
 'EX_glc__D_e': 10.0,
 'EX_h_e': 1000.0,
 'EX_h2o_e': 1000.0,
 'EX_k_e': 1000.0,
 'EX_cu2_e': 1000.0,
 'EX_mg2_e': 1000.0,
 'EX_mn2_e': 1000.0,
 'EX_mobd_e': 1000.0,
 'EX_na1_e': 1000.0,
 'EX_nh4_e': 1000.0,
 'EX_ca2_e': 1000.0,
 'EX_cbl1_e': 0.01,
 'EX_ni2_e': 1000.0,
 'EX_o2_e': 1000.0,
 'EX_cl_e': 1000.0,
 'EX_pi_e': 1000.0,
 'EX_zn2_e': 1000.0,
 'EX_sel_e': 1000.0,
 'EX_slnt_e': 1000.0,
 'EX_so4_e': 1000.0,
 'EX_tungs_e': 1000.0,
 'EX_fe2_e': 1000.0,
 'EX_fe3_e': 1000.0,
 'EX_eth_e': 1000.0}

Note, that the model already has plenty of trace elements. It's also running 'aerobic' because 'EX_o2_e': 1000.0 

In [14]:
#model.reactions #Used during coding, but not needed in final version

This code is compiled to figure out the initial benchmark in terms of Growth-rate and EFE_m reaction productivity.

In [15]:

with model:
    solution = model.optimize()
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical growth-rate: 0.9823718127269712 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 12.780573951434894 mmol/gDW*h


2 Level, 21 Factor, Fractorial DOE is performed (2^(21-16)) to screen for the amino acids (AA) which had the biggest influence on growth-rate and EFE_m productivity. Glucose is used as C-source at level 20 or 40. All 20 common AA are used at level 0 or 20.

Result:
Out of 20 AA, these 13 have the biggest influence:
Ala
Arg
Asn
Asp
Cys
Gln
Glu
Pro
Ser
Thr
Trp


Glucose at 40 is better then at 20

Maximum theoretical growth-rate was increased from 0.982 /h to 17.3 /h in the 'Statistically best' run. This is roughly a 18-times improvement.
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction was increased from 12.8 mmol/gDW*h to 225.4 mmol/gDW*h. This is roughly a 18 times improvement.

By changing the C-source from Glucose to Sucrose Maximum theoretical growth-rate was increased from 0.982 /h to 19.2 /h in the 'Statistically best' run. This is roughly a 20-times improvement.
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction was increased from 12.8 mmol/gDW*h to 277 mmol/gDW*h. This is roughly a 22-times improvement.

Making Media for Run 1 of 32

In [16]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

Productivity using Run 1 Media

In [17]:
with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical growth-rate: 9.278337088982058 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 99.12641509433922 mmol/gDW*h


Making Media for Run 2 of 32

In [18]:
YEPDmedium2 = model.medium
YEPDmedium2['EX_glc__D_e'] = 20
YEPDmedium2['EX_ala__L_e'] = 0
YEPDmedium2['EX_arg__L_e'] = 0
YEPDmedium2['EX_asn__L_e'] = 0
YEPDmedium2['EX_cys__L_e'] = 20
YEPDmedium2['EX_gln__L_e'] = 20
YEPDmedium2['EX_glu__L_e'] = 0
YEPDmedium2['EX_gly_e'] = 0
YEPDmedium2['EX_his__L_e'] = 20
YEPDmedium2['EX_ile__L_e'] = 0
YEPDmedium2['EX_leu__L_e'] = 20
YEPDmedium2['EX_met__L_e'] = 20
YEPDmedium2['EX_phe__L_e'] = 0
YEPDmedium2['EX_pro__L_e'] = 0
YEPDmedium2['EX_ser__L_e'] = 20
YEPDmedium2['EX_thr__L_e'] = 20
YEPDmedium2['EX_trp__L_e'] = 0
YEPDmedium2['EX_tyr__L_e'] = 20
YEPDmedium2['EX_val__L_e'] = 0
YEPDmedium2['EX_lys__L_e'] = 0
YEPDmedium2['EX_asp__L_e'] = 20

Productivity using Run 2 Media

In [19]:
with model:
    model.medium = YEPDmedium2
    solution = model.optimize()
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical growth-rate: 8.15391882498941 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 98.18301886792437 mmol/gDW*h


Making Media for Run 3

In [20]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

Productivity for Run 3 Media

In [21]:
with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Maximum theoretical growth-rate: 8.394074883344196 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 92.14528301886806 mmol/gDW*h


Media and Productivity for Run 4

In [22]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 4:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 4:
Maximum theoretical growth-rate: 8.578553785827065 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 105.16415094339632 mmol/gDW*h


Media and Productivity Run 5:

In [23]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 5:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 5:
Maximum theoretical growth-rate: 9.123156211139484 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 98.18301886792426 mmol/gDW*h


Media and Productivity for Run 6:

In [24]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 6:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 6:
Maximum theoretical growth-rate: 10.044916712188886 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 118.74905660377382 mmol/gDW*h


Media and Productivity for Run 7:

In [25]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 7:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 7:
Maximum theoretical growth-rate: 11.474637602157413 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 131.9566037735842 mmol/gDW*h


Media and Productivity for Run 8:

In [26]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 8:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 8:
Maximum theoretical growth-rate: 11.83992043142593 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 138.5603773584906 mmol/gDW*h


Media and Productivity for Run 9:

In [27]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 9:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 9:
Maximum theoretical growth-rate: 7.715016342916728 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 86.86226415094346 mmol/gDW*h


Media and Productivity for Run 10:

In [28]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 10:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 10:
Maximum theoretical growth-rate: 9.588258989523565 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 107.0509433962263 mmol/gDW*h


Media and Productivity for Run 11:

In [29]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 11:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 11:
Maximum theoretical growth-rate: 9.230116382842363 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 106.67358490566053 mmol/gDW*h


Media and Productivity for Run 12:

In [30]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 12:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 12:
Maximum theoretical growth-rate: 12.263779174674635 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 140.82452830188606 mmol/gDW*h


Media and Productivity for Run 13:

In [31]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 13:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 13:
Maximum theoretical growth-rate: 12.03192642121634 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 133.84339622641534 mmol/gDW*h


Media and Productivity for Run 14:

In [32]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 14:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 14:
Maximum theoretical growth-rate: 11.718214527486467 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 133.27735849056668 mmol/gDW*h


Media and Productivity for Run 15:

In [33]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 15:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 15:
Maximum theoretical growth-rate: 12.10438962928361 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 140.82452830188754 mmol/gDW*h


Media and Productivity for Run 16:

In [34]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 20
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 16:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 16:
Maximum theoretical growth-rate: 11.202610270549965 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 126.29622641509427 mmol/gDW*h


Media and Productivity for Run 17:

In [35]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 17:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 17:
Maximum theoretical growth-rate: 10.893551562846541 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 122.43968253968279 mmol/gDW*h


Media and Productivity for Run 18:

In [36]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 18:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 18:
Maximum theoretical growth-rate: 9.165181952757557 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 100.39500000000012 mmol/gDW*h


Media and Productivity for Run 19:

In [37]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 19:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 19:
Maximum theoretical growth-rate: 10.346251307554406 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 114.02698412698402 mmol/gDW*h


Media and Productivity for Run 20:

In [38]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 20:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 20:
Maximum theoretical growth-rate: 13.478264575343301 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 162.52264150943392 mmol/gDW*h


Media and Productivity for Run 21:

In [39]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 21:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 21:
Maximum theoretical growth-rate: 11.782936409853656 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 127.2396226415088 mmol/gDW*h


Media and Productivity for Run 22:

In [40]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 22:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 22:
Maximum theoretical growth-rate: 14.67463709251063 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 168.9377358490569 mmol/gDW*h


Media and Productivity for Run 23:

In [41]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 23:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 23:
Maximum theoretical growth-rate: 14.419707338907115 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 162.52264150943407 mmol/gDW*h


Media and Productivity for Run 24:

In [42]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 0
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 24:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 24:
Maximum theoretical growth-rate: 11.467825937991956 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 133.65471698113225 mmol/gDW*h


Media and Productivity for Run 25:

In [43]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 25:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 25:
Maximum theoretical growth-rate: 12.496338974960711 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 136.67358490566045 mmol/gDW*h


Media and Productivity for Run 26:

In [44]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 26:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 26:
Maximum theoretical growth-rate: 12.050744002953337 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 136.48490566037725 mmol/gDW*h


Media and Productivity for Run 27:

In [45]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 27:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 27:
Maximum theoretical growth-rate: 14.2849187450899 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 185.9188679245285 mmol/gDW*h


Media and Productivity for Run 28:

In [46]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 0
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 28:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 28:
Maximum theoretical growth-rate: 8.885559238096986 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 87.06166666666667 mmol/gDW*h


Media and Productivity for Run 29:

In [47]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 29:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 29:
Maximum theoretical growth-rate: 11.945371978116018 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 136.48490566037745 mmol/gDW*h


Media and Productivity for Run 30:

In [48]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 0
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 0

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 30:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 30:
Maximum theoretical growth-rate: 14.591504738742723 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 156.29622641509368 mmol/gDW*h


Media and Productivity for Run 31:

In [49]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 0
YEPDmedium['EX_gln__L_e'] = 0
YEPDmedium['EX_glu__L_e'] = 0
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 0
YEPDmedium['EX_ser__L_e'] = 0
YEPDmedium['EX_thr__L_e'] = 0
YEPDmedium['EX_trp__L_e'] = 0
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 31:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 31:
Maximum theoretical growth-rate: 8.609986872263448 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 114.02698412698449 mmol/gDW*h


Media and Productivity for Run 32:

In [50]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 20
YEPDmedium['EX_ile__L_e'] = 20
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 20
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Run 32:')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Run 32:
Maximum theoretical growth-rate: 19.36915342869881 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 232.33396226415047 mmol/gDW*h


Testing 'Statistically Best' Media:

In [51]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best (7missing):')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best:
Maximum theoretical growth-rate: 17.792447707656347 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 232.33396226415093 mmol/gDW*h


Statistically best best:

In [55]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 20
YEPDmedium['EX_met__L_e'] = 20
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 20
YEPDmedium['EX_val__L_e'] = 20
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best (4missing):')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best (4missing):
Maximum theoretical growth-rate: 18.251778743397885 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 232.33396226415087 mmol/gDW*h


In [54]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best (9missing):')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best (9missing):
Maximum theoretical growth-rate: 17.292327443315337 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 225.35283018867912 mmol/gDW*h


In [56]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 20
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 20
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best (7missing):')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best (7missing):
Maximum theoretical growth-rate: 17.710013295256736 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 232.33396226415002 mmol/gDW*h


Now we test the Statistically best media (9 missing) with the two alternative C-sources: Sucrose and Fructose at 40.

In [59]:
model.reactions.EX_sucr_e

<Reaction EX_sucr_e at 0x7f3cddd5c370>

In [60]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 0
YEPDmedium['EX_sucr_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best (With Sucrose instead of Glucose):')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best (9 missing):
Maximum theoretical growth-rate: 19.19147450095909 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 277.0509433962263 mmol/gDW*h


In [62]:
model.reactions.EX_fru_e

<Reaction EX_fru_e at 0x7f3cddbfd9f0>

In [63]:
YEPDmedium = model.medium
YEPDmedium['EX_glc__D_e'] = 0
YEPDmedium['EX_sucr_e'] = 0
YEPDmedium['EX_fru_e'] = 40
YEPDmedium['EX_ala__L_e'] = 20
YEPDmedium['EX_arg__L_e'] = 20
YEPDmedium['EX_asn__L_e'] = 20
YEPDmedium['EX_cys__L_e'] = 20
YEPDmedium['EX_gln__L_e'] = 20
YEPDmedium['EX_glu__L_e'] = 20
YEPDmedium['EX_gly_e'] = 0
YEPDmedium['EX_his__L_e'] = 0
YEPDmedium['EX_ile__L_e'] = 0
YEPDmedium['EX_leu__L_e'] = 0
YEPDmedium['EX_met__L_e'] = 0
YEPDmedium['EX_phe__L_e'] = 0
YEPDmedium['EX_pro__L_e'] = 20
YEPDmedium['EX_ser__L_e'] = 20
YEPDmedium['EX_thr__L_e'] = 20
YEPDmedium['EX_trp__L_e'] = 20
YEPDmedium['EX_tyr__L_e'] = 0
YEPDmedium['EX_val__L_e'] = 0
YEPDmedium['EX_lys__L_e'] = 0
YEPDmedium['EX_asp__L_e'] = 20

with model:
    model.medium = YEPDmedium
    solution = model.optimize()
    print('Statistically best (9 missing):')
    print('Maximum theoretical growth-rate:', solution.fluxes['BIOMASS_Ec_iJO1366_core_53p95M'], '/h')

    model.objective = model.reactions.EFE_m
    E_production = model.optimize().objective_value
    print('Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction:', E_production, 'mmol/gDW*h')


Statistically best (9 missing):
Maximum theoretical growth-rate: 17.292327443298877 /h
Maximum theoretical productivity of Ethylene_forming_enzyme_main reaction: 224.22075471698122 mmol/gDW*h
